src/metabase/query_processor/middleware/constraints.clj

```diff
@@ -5,33 +5,46 @@
+(def ^:private ^:const default-max-unaggregated-query-row-limit 2000)
+(def ^:private ^:const default-max-aggregated-query-row-limit 10000)
 
-;; NOTE: this was changed from a hardcoded var with value of 2000 (now moved to [[default-max-results-bare-rows]])
+;; NOTE: this was changed from a hardcoded var with value of 2000 (now moved to [[default-max-unaggregated-query-row-limit]])
 ;; to a setting in 0.43 the setting, which allows for DB local value, can still be nil, so any places below that used
 ;; to reference the former constant value have to expect it could return nil instead
-(setting/defsetting max-results-bare-rows
+(setting/defsetting max-unaggregated-query-row-limit
   (deferred-tru "Maximum number of rows to return specifically on :rows type queries via the API.")
   :visibility     :authenticated
   :type           :integer
   :database-local :allowed
   :audit          :getter)
 
-(def ^:private max-results
-  "General maximum number of rows to return from an API query."
-  10000)
+(setting/defsetting max-aggregated-query-row-limit
+  (deferred-tru "Maximum number of rows to return for aggregated queries via the API.")
+  :visibility     :authenticated
+  :type           :integer
+  :database-local :allowed
+  :audit          :getter)
+
+(defn query->max-rows
+  "Given a query, returns the max rows that should be returned *as defined by settings*. In other words,
+  return `(max-aggregated-query-row-limit)` or `(max-unaggregated-query-row-limit)` depending on whether the query is
+  aggregated or not."
+  [{{aggregations :aggregation} :query}]
+  (if-not aggregations
+    (max-unaggregated-query-row-limit)
+    (max-aggregated-query-row-limit)))
 
 (defn default-query-constraints
   "Default map of constraints that we apply on dataset queries executed by the api."
   []
-  {:max-results           max-results
-   :max-results-bare-rows (or (max-results-bare-rows) default-max-results-bare-rows)})
+  {:max-results           (or (max-aggregated-query-row-limit) default-max-aggregated-query-row-limit)
+   :max-results-bare-rows (or (max-unaggregated-query-row-limit) default-max-unaggregated-query-row-limit)})
 
 (defn- ensure-valid-constraints
-  "`:max-results-bare-rows` must be less than or equal to `:max-results`, so if someone sets `:max-results` but not
-  `:max-results-bare-rows` or sets an both but sets an invalid value for ``:max-results-bare-rows` use the same value
-  for both. Otherwise the default bare rows value could end up being higher than the custom `:max-rows` value, causing
-  an error."
+  "`:max-unaggregated-query-row-limit` must be less than or equal to `:max-results`, so if someone sets `:max-results`
+  but not `:max-unaggregated-query-row-limit` or sets an both but sets an invalid value for
+  ``:max-unaggregated-query-row-limit` use the same value for both. Otherwise the default bare rows value could end up
+  being higher than the custom `:max-rows` value, causing an error."
```

Comment:
I'm not sure this is clearer than original comment.

Reply:
I didn't change the comment itself, only changed `max-results-bare-rows` to `max-unaggregated-query-row-limit`. I'm happy to reword it though, I find the comment a bit confusing too.

Reply:
Yes, so it's referring to things that don't appear in the code. I think the connection between these things should be explained.

Reply:
Oh, great catch - sorry, I don't know how I missed that.

Reply:
After taking a step back, I just reworded the comment to ["Clamps the value of `max-results-bare-rows` to be less than or equal to the value of `max-results`."](https://github.com/metabase/metabase/pull/35600/files#diff-3b2ac9a54646c3fc99a8239c4ae27cbc1478f01837a6e423c6a8b96a1a32c6e5R60)

I don't *think* it's necessary to point out the names of the settings those values actually originate from (and the code pulling them from the settings is in the same short file, so it's pretty easy to trace), but let me know if you'd rather include that elaboration and I'll add that in.